codegenerator/cli/templates/static/codegen/src/UserContext.res

```diff
@@ -208,70 +204,128 @@ let getWhereTraps: Utils.Proxy.traps<loaderEntityContextParams> = {
   },
 }
 
-let makeEntityLoaderContext = (params): Types.entityLoaderContext<
-  Entities.internalEntity,
-  unknown,
-> => {
-  let get = entityId =>
-    params.loadLayer->LoadLayer.loadById(
-      ~entityConfig=params.entityConfig,
-      ~inMemoryStore=params.inMemoryStore,
-      ~shouldGroup=params.shouldGroup,
-      ~eventItem=params.eventItem,
-      ~entityId,
-    )
-  {
-    get,
-    getOrThrow: (entityId, ~message=?) =>
-      get(entityId)->Promise.thenResolve(entity => {
-        switch entity {
-        | Some(entity) => entity
-        | None =>
-          Js.Exn.raiseError(
-            message->Belt.Option.getWithDefault(
-              `Entity '${params.entityConfig.name}' with ID '${entityId}' is expected to exist.`,
+let noopSet = (_entity: Internal.entity) => ()
+
+let entityTraps: Utils.Proxy.traps<entityContextParams> = {
+  get: (~target as params, ~prop: unknown) => {
+    let prop = prop->(Utils.magic: unknown => string)
+
+    let set = params.isPreload
+      ? noopSet
+      : (entity: Internal.entity) => {
+          params.inMemoryStore
+          ->InMemoryStore.getInMemTable(~entityConfig=params.entityConfig)
+          ->InMemoryTable.Entity.set(
+            Set(entity)->Types.mkEntityUpdate(
+              ~eventIdentifier=params.eventItem->makeEventIdentifier,
+              ~entityId=entity.id,
             ),
+            ~shouldSaveHistory=params.shouldSaveHistory,
```

Comment:
Should this be inlined so that it's only constructed if set is accessed?

Reply:
@inline doesn't work for function inside of another function. I also tried it first :)